src/app/utils/DiscordUtils.ts

```diff
@@ -190,15 +190,29 @@ const DiscordUtils = {
     },
 
     // Send a notification to an interested party (use a DM)
-    async activityNotification(content: string, toUser: GuildMember, link: string): Promise<void> {
+    async activityNotification(content: string, toUser: GuildMember, link: string, messageEmbeds?: MessageEmbedOptions): Promise<void> {
         try {
-            const componentActions = new MessageActionRow().addComponents(
-                new MessageButton()
+            const componentAction = [];
+            const linkButton = new MessageButton()
                     .setLabel('View Bounty')
                     .setStyle('LINK')
-                    .setURL(link || '')
-            );
-            await toUser.send({ content, components: link && [componentActions] });
+                    .setURL(link || '');
+
+            if (messageEmbeds) {
```

Comment:
Can we build this outside of the activityNotification routine? It is adding activity-specific logic into a utility. I think you could add the claim button in Assign.ts and keep the utility free of activity-specific knowledge.